src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1030,53 +1034,71 @@ static bool isOrContainsErrorType(TypeSymbol typeSymbol)
+                                                 checkParameters ?
+                                                     (diagnostics, overriddenMethod, overridingMethod, overridingParameter, location) =>
+                                                     {
+                                                         diagnostics.Add(
+                                                             ErrorCode.WRN_NullabilityMismatchInParameterTypeOnOverride,
+                                                             location,
+                                                             new FormattedSymbol(overridingParameter, SymbolDisplayFormat.ShortFormat));
+                                                     } :
+                                                     (Action<DiagnosticBag, MethodSymbol, MethodSymbol, ParameterSymbol, Location>)null,
+                                                 overridingMemberLocation);
+            }
+        }
 
-                var conversions = compilation.Conversions.WithNullability(true);
-                if (!isValidNullableConversion(
-                        conversions,
-                        overridingMethod.RefKind,
-                        overridingMethod.ReturnTypeWithAnnotations,
-                        overriddenMethod.ReturnTypeWithAnnotations))
-                {
-                    diagnostics.Add(ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverride, overridingMemberLocation);
-                    return;
-                }
+        internal static void CheckValidNullableMethodOverride<TArg>(
+            CSharpCompilation compilation,
+            MethodSymbol overriddenMethod,
+            MethodSymbol overridingMethod,
+            DiagnosticBag diagnostics,
+            Action<DiagnosticBag, MethodSymbol, MethodSymbol, TArg> reportMismatchInReturnType,
+            Action<DiagnosticBag, MethodSymbol, MethodSymbol, ParameterSymbol, TArg> reportMismatchInParameterType,
+            TArg extraArgument)
+        {
+            if (!CheckValidNullableOverride(compilation, overriddenMethod, overridingMethod))
+            {
+                return;
+            }
 
-                if (!checkParameters)
-                {
-                    return;
-                }
+            var conversions = compilation.Conversions.WithNullability(true);
+            if (reportMismatchInReturnType != null &&
+                !isValidNullableConversion(
+                    conversions,
+                    overridingMethod.RefKind,
+                    overridingMethod.ReturnTypeWithAnnotations,
+                    overriddenMethod.ReturnTypeWithAnnotations))
+            {
+                reportMismatchInReturnType(diagnostics, overriddenMethod, overridingMethod, extraArgument);
+                return;
+            }
+
+            if (reportMismatchInParameterType == null)
```

Comment:
Consider moving this `if` before the previous one, and removing the `reportMismatchInParameterType != null` check from that one.

Reply:
Wait, ignore. Misread parameter for return.

---
In reply to: [296389113](https://github.com/dotnet/roslyn/pull/36663#discussion_r296389113) [](ancestors = 296389113)